src/core/infer_response.cc

```diff
@@ -73,6 +73,12 @@ InferenceResponse::AddOutput(
 
   LOG_VERBOSE(1) << "add response output: " << outputs_.back();
 
+  if (backend_ != nullptr) {
```

Comment:
backend_raw_ is guaranteed to be non-null so you should use that.

Reply:
InferenceResponse class only has `backend_`